lib/passenger.rb

```diff
@@ -16,6 +16,18 @@ def add_trip(trip)
       @trips << trip
     end
 
+    def net_expenditures
+      raise ArgumentError.new("no trips found for this passenger") if @trips.empty?
+      complete_trips = @trips.find_all { |trip| trip.end_time != nil }
+      return complete_trips.sum { |trip| trip.cost }
+    end
+
+    def total_time_spent
+      raise ArgumentError.new("no trips found for this passenger") if @trips.empty?
+      complete_trips = @trips.find_all { |trip| trip.end_time != nil }
+      return complete_trips.sum { |trip| trip.duration }
+    end
```

Comment:
so I wouldn't actually call this a good situation for an exception! things like `ArgumentError` tend to indicate that something broken happened, or a bad state was reached. It's totally normal for a passenger to have never used a service before! Instead, the thing to do would to be to return 0 in those situations.